bin/mocha.js

```diff
@@ -109,7 +110,13 @@ if (mochaArgs['node-option'] || Object.keys(nodeArgs).length || hasInspect) {
   proc.on('exit', (code, signal) => {
     process.on('exit', () => {
       if (signal) {
-        process.kill(process.pid, signal);
+        const numericSignal =
+          typeof signal === 'string' ? os.constants.signals[signal] : signal;
+        if (mochaArgs['posix-exit-codes'] === true) {
+          process.exit(128 + numericSignal);
+        } else {
+          process.kill(process.pid, signal);
+        }
       } else {
         process.exit(code);
```

Comment:
[Bug] Sorry, I might have missed this in my first review: when I run the "hello world" example from https://github.com/mochajs/mocha-examples/pull/73, `signal` is `null` and `code` is the number of failed tests. Modifying the tests to fail 49 and pass 1:

```js
describe("example", () => {
  for (let i = 0; i < 50; i += 1) {
    it("2 + 2 (" + i + ")", () => {
      assert.equal(2 + 2, i);
    });
  }
});
```

...then `code` is `49` and `signal` is `null`.

Shouldn't the `posix-exit-codes` logic also handle the case of `process.exit(code)`? Am I missing something?

```suggestion
        process.exit(code + (mochaArgs['posix-exit-codes'] ? 128 : 0));
```

Note that my suggestion snippet is only part of the way there. There's also a complication of having over 127 (255 - 128) tests. That exit code should be capped to 255, I think. Fun fact: `node -e "process.exit(270)"; echo $?` reports an exit code of 14 (270 % 256).

Reply:
The posix exit codes are a numerical representation of the signal that terminated the process, so if the process ran normally and wasn't pre-empted by some signal, the posix-compliant exit code of the process would be `0`.  In this case, no signal was raised, so the choice was between exiting with code of `0` or continuing with the legacy behavior of exiting with the number of failed tests. I chose to continue the legacy behavior but looking at this again now, I think the correct response would be to: 
```
process.exit(mochaArgs['posix-exit-codes'] ? 0 : code);
```